### PR TITLE
Ensure error status codes are surfaced from client failures

### LINKED ISSUE #1574: Passing status code back from executor service orchestrator
The current executor code passes back the payload received on error from downstream components but sets the statusCode to 500. We should look to pass back the statusCode for downstream REST components.

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #1574 

 * Add custom golang error to pass status code back to top level
 * Adds status code and content type as passed back by client
 * Fixes python test for non 200 use case
 * Cleans up test server struct

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Mar 23 07:48:19 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=2

**Comment by seldondev** [comment]
Mon Mar 23 07:48:25 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=1

**Comment by seldondev** [comment]
Tue Mar 24 14:53:45 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=3

**Comment by seldondev** [comment]
Tue Mar 24 14:53:48 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=4

**Comment by seldondev** [comment]
Tue Mar 24 14:53:58 UTC 2020
The logs for [integration] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=5

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 28dda1e6b20f72a45b6d62f72bf6537e34f0632e | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: executor/api/test/seldonmessage_test_client.go [hunk 1]
```diff
@@ -2,20 +2,18 @@ package test
...
 
 type SeldonMessageTestClient struct {
-	t           *testing.T
-	chosenRoute int
-	errMethod   *v1.PredictiveUnitMethod
-	err         error
+	ChosenRoute int
+	ErrMethod   *v1.PredictiveUnitMethod
+	Err         error
+	ErrPayload  payload.SeldonPayload
```
Comment (adriangonz) [inline executor/api/test/seldonmessage_test_client.go]: Do we need the `ErrPayload` field? It doesn't seem to be used on the PR.

**Comment by seldondev** [comment]
Tue Mar 24 17:12:49 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=6

**Comment by seldondev** [comment]
Tue Mar 24 17:12:50 UTC 2020
The logs for [lint] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=7

File: executor/api/test/seldonmessage_test_client.go (hunk 1 above)
Comment (cliveseldon) [reply executor/api/test/seldonmessage_test_client.go]: updated a test to use it

File: executor/api/grpc/seldon/server_test.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func TestPredict(t *testing.T) {
 		},
 	}
 	url, _ := url.Parse("http://localhost")
-	server := NewGrpcSeldonServer(&p, test.NewSeldonMessageTestClient(t, 0, nil, nil), url, "default")
+	server := NewGrpcSeldonServer(&p, test.SeldonMessageTestClient{}, url, "default")
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply executor/api/grpc/seldon/server_test.go]: done

**Comment by seldondev** [comment]
Tue Mar 24 17:34:15 UTC 2020
The logs for [pr-build] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1575/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1575 --build=8

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1575#issuecomment-603987431" title="Approved">adriangonz</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**